src/main/java/frc/robot/commands/Vision/SetGoalRPM.java

```diff
@@ -5,29 +5,42 @@
 package frc.robot.commands.Vision;
 
 import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.RobotPreferences.VisionPrefs;
+import frc.robot.subsystems.Hood;
 import frc.robot.subsystems.Shooter;
 import frc.robot.subsystems.Vision;
 
 public class SetGoalRPM extends CommandBase {
   Shooter shooter;
   Vision vision;
+  Hood hood;
 
   /** Creates a new SetGoalRPM. */
-  public SetGoalRPM(Shooter sub_shooter, Vision sub_vision) {
+  public SetGoalRPM(Shooter sub_shooter, Vision sub_vision, Hood sub_hood) {
     // Use addRequirements() here to declare subsystem dependencies.
     shooter = sub_shooter;
     vision = sub_vision;
+    hood = sub_hood;
   }
 
   // Called when the command is initially scheduled.
   @Override
   public void initialize() {
+    if (vision.limelightDistanceFromGoal() < VisionPrefs.lowHoodMaxDistance.getValue()) {
+      hood.hoodLowTilt();
+      shooter.setGoalRPM(vision.limelightLowDistanceRPM());
+    } else if (vision.limelightDistanceFromGoal() < VisionPrefs.midHoodMaxDistance.getValue()) {
+      hood.hoodMediumTilt();
+      shooter.setGoalRPM(vision.limelightMidDistanceRPM());
+    } else if (vision.limelightDistanceFromGoal() < VisionPrefs.highHoodMaxDistance.getValue()) {
+      hood.hoodHighTilt();
+      shooter.setGoalRPM(vision.limelightHighDistanceRPM());
+    }
```

Comment:
there isn't any behavior for if the limelight doesn't have a target. that should be defined and do something (like maybe preset to fender?)